core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -111,6 +115,40 @@ abstract class BaseConsumerTest extends AbstractConsumerTest {
 }
 
 object BaseConsumerTest {
+  // We want to test the following combinations:
+  // * ZooKeeper and the generic group protocol
+  // * KRaft and the generic group protocol
+  // * KRaft with the new group coordinator enabled and the consumer group protocol
+  def getTestQuorumAndGroupProtocolParametersAll() : java.util.stream.Stream[Arguments] = {
+    java.util.stream.Stream.of(
+        Arguments.of("zk", "generic"),
+        Arguments.of("kraft", "generic"),
+        Arguments.of("kraft+kip848", "consumer"))
+  }
+
+  // We want to test the following combinations:
+  // * ZooKeeper and the generic group protocol
+  def getTestQuorumAndGroupProtocolParametersZkOnly() : java.util.stream.Stream[Arguments] = {
+    java.util.stream.Stream.of(
+        Arguments.of("zk", "generic"))
+  }
+
+  // For tests that only work with the generic group protocol, we want to test the following combinations:
+  // * ZooKeeper and the generic group protocol
+  // * KRaft and the generic group protocol
+  def getTestQuorumAndGroupProtocolParametersGenericGroupProtocolOnly() : java.util.stream.Stream[Arguments] = {
+    java.util.stream.Stream.of(
+        Arguments.of("zk", "generic"),
+        Arguments.of("kraft", "generic"))
```

Comment:
We likely need it here too.

Reply:
OK. Nice and easy to change now I've refactored it. I'll get on it :)